### PR TITLE
`ProtocConfigurationPlugin` tuning for pure JS projects

### PR DESCRIPTION
**Pull request by serhii-lekariev**
This PR brings changes necessary to work on https://github.com/SpineEventEngine/bootstrap/issues/16.

Before `ProtocConfigurationPlugin` has managed a lot of configurations useful for both Java and JavaScript projects, however, it could not be used for JS projects. Since it was tightly coupled with Spine Model Compiler plugin, which is only applicable to Java projects.

This PR decouples the abstract base away from the Model Compiler plugin, substituting it with an extension tuned specifically for Java projects.

As a result, an abstract `ProtocConfigurationPlugin` is now usable outside of Model Compiler, allowing to reuse its configuration logic.

### CODE REVIEW

**Comment by serhii-lekariev** [comment]
@dmdashenkov, PTAL.

File: tools/plugin-base/src/main/java/io/spine/tools/gradle/ProtocConfigurationPlugin.java [hunk 1]
```diff
@@ -92,7 +73,8 @@
     @Override
     public void apply(Project project) {
         project.getPluginManager()
-               .withPlugin(ProtobufDependencies.gradlePlugin().value(), plugin -> applyTo(project));
+               .withPlugin(ProtobufDependencies.gradlePlugin()
+                                               .value(), plugin -> applyTo(project));
```
Comment (dmdashenkov) [inline tools/plugin-base/src/main/java/io/spine/tools/gradle/ProtocConfigurationPlugin.java]: Please align the params more neatly.

File: tools/plugin-base/src/main/java/io/spine/tools/gradle/ProtocConfigurationPlugin.java [hunk 2]
```diff
@@ -304,6 +225,10 @@ private static boolean isTestsTask(GenerateProtoTask protocTask) {
         ProtocPlugin(String name) {
             this.name = name;
         }
+
+        public String getName() {
```
Comment (dmdashenkov) [inline tools/plugin-base/src/main/java/io/spine/tools/gradle/ProtocConfigurationPlugin.java]: Please avoid `get` prefixes.

File: tools/proto-js-plugin/src/main/java/io/spine/js/gradle/ProtocConfigurationPluginForJs.java [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+import org.gradle.api.Task;
+
+import java.io.File;
+import java.nio.file.Path;
+
+/**
+ * A Gradle plugin that performs additional {@code protoc} configurations relevant for JavaScript
+ * projects.
+ */
+public final class ProtocConfigurationPluginForJs extends ProtocConfigurationPlugin {
```
Comment (dmdashenkov) [inline tools/proto-js-plugin/src/main/java/io/spine/js/gradle/ProtocConfigurationPluginForJs.java]: I suggest `[Js|Java]ProtocConfigPlugin`. We don't need an extra word in there.

**Comment by serhii-lekariev** [comment]
@dmdashenkov, PTAL again.

**Comment by serhii-lekariev** [comment]
@dmdashenkov, PTAL again.

File: tools/plugin-base/src/main/java/io/spine/tools/gradle/ProtocConfigurationPlugin.java [hunk 4]
```diff
@@ -201,109 +182,41 @@ private void configureDescriptorSetGeneration(GenerateProtoTask protocTask) {
...
-        GRPC("grpc"),
-        SPINE("spineProtoc");
-
-        @SuppressWarnings("PMD.SingularField") /* Accessed from the outer class. */
-        private final String name;
-
-        ProtocPlugin(String name) {
-            this.name = name;
-        }
+    protected enum ProtocPlugin {
```
Comment (dmdashenkov) [inline tools/plugin-base/src/main/java/io/spine/tools/gradle/ProtocConfigurationPlugin.java]: Please document

**Comment by serhii-lekariev** [comment]
@dmdashenkov, PTAL again.

**Pull request merged**